### PR TITLE
MON-1679: adding static auth to KSM

### PR DESCRIPTION
**Pull request by prashbnair**
* [ ] I added CHANGELOG entry for this change.
* [ ] No user facing changes, so no entry in CHANGELOG was needed.

### CODE REVIEW

**Comment by prashbnair** [comment]
@jan--f @s-urbaniak  I have implemented the changes for KSM and if this looks good, I can implement it for other components.

**Comment by simonpasquier** [comment]
lgtm
Have you tried it with https://github.com/openshift/kube-rbac-proxy/pull/51 to make sure that it works without the hardcoded authorizer?

**Comment by simonpasquier** [comment]
Looking at https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_cluster-monitoring-operator/1318/pull-ci-openshift-cluster-monitoring-operator-master-e2e-agnostic-operator/1426158852574810112 it seems that the e2e operator tests are failing because we hit the 30 minutes timeout...

**Comment by prashbnair** [comment]
> Have you tried it with openshift/kube-rbac-proxy#51 to make sure that it works without the hardcoded authorizer?

Yes

**Comment by prashbnair** [comment]
> Looking at https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_cluster-monitoring-operator/1318/pull-ci-openshift-cluster-monitoring-operator-master-e2e-agnostic-operator/1426158852574810112 it seems that the e2e operator tests are failing because we hit the 30 minutes timeout...

This does not seem related to my change, the test that is failing is for thanos querier and I have not made any changes in that area.

**Comment by jan--f** [comment]
The approach looks good I think. Maybe the secret could have a more descriptive name, i.e. mention static auth for the metric path?

**Comment by simonpasquier** [comment]
> This does not seem related to my change, the test that is failing is for thanos querier and I have not made any changes in that area.

Correct I wasn't suggesting that it was introduced by your PR.

File: jsonnet/components/kube-state-metrics.libsonnet [hunk 1]
```diff
@@ -72,6 +72,34 @@ function(params)
       },
     },
 
+    kubeRbacProxySecret: {
+      apiVersion: 'v1',
+      kind: 'Secret',
+      metadata: {
+        name: 'kube-state-metrics-kube-rbac-proxy',
```
Comment (simonpasquier) [inline jsonnet/components/kube-state-metrics.libsonnet:79]: As @jan--f suggested, the secret could be more descriptive 

```suggestion
        name: 'kube-state-metrics-kube-rbac-proxy-config',
```

**Comment by prashbnair** [comment]
/retitle MON-1679 adding static auth to KSM

File: jsonnet/components/kube-state-metrics.libsonnet (hunk 1 above)
Comment (prashbnair) [reply jsonnet/components/kube-state-metrics.libsonnet]: Done

File: jsonnet/components/kube-state-metrics.libsonnet [hunk 2]
```diff
@@ -107,6 +136,11 @@ function(params)
                           name: 'metrics-client-ca',
                           readOnly: false,
                         },
+                        {
+                          mountPath: '/etc/kube-rbac-policy',
+                          name: 'kube-state-metrics-kube-rbac-proxy-config',
+                          readOnly: false,
```
Comment (simonpasquier) [inline jsonnet/components/kube-state-metrics.libsonnet:142]: (nit)

```suggestion
                          readOnly: true,
```

**Review by simonpasquier — COMMENTED** [review]
/hold waiting for 4.10
LGTM, I think you can go ahead with the other components. Maybe there' some way to have jsonnet functions to avoid code duplication?

**Comment by prashbnair** [comment]
Added jsonnet function to generate the secret.

**Review by simonpasquier — COMMENTED** [review]
Can you do the same change for the other components?
* node-exporter
* openshift-state-metrics
* prometheus-operator
* prometheus-operator-user-workload
* prometheus-user-workload
* telemeter-client

**Comment by prashbnair** [comment]
Can we do the changes in stages, once this PR is merged I will start work on the other components and raise PR's for each of them.

**Comment by simonpasquier** [comment]
sure, lets do it in stages.

/lgtm

/label px-approved
/label docs-approved
/label qe-approved

The change has no user-facing impact and the CI should fail if the metrics collection is broken.

**Pull request merged**